src/test/scala/org/apache/spark/deploy/armada/submit/ArmadaClientApplicationSuite.scala

```diff
@@ -0,0 +1,906 @@
+    config.driverJobItemTemplate should not be empty
+    config.driverJobItemTemplate.get.namespace shouldBe "all-driver-namespace"
+    config.driverJobItemTemplate.get.priority shouldBe 1.5
+    config.driverJobItemTemplate.get.labels should contain(
+      "item-template-label" -> "item-template-value"
+    )
+    config.executorJobItemTemplate should not be empty
+    config.executorJobItemTemplate.get.namespace shouldBe "all-executor-namespace"
+    config.executorJobItemTemplate.get.priority shouldBe 2.5
+    config.executorJobItemTemplate.get.labels should contain(
+      "item-template-label" -> "item-template-value"
+    )
+    config.cliConfig.queue shouldBe Some("test-queue")
+    config.cliConfig.jobSetId shouldBe Some("test-job-set")
+  }
+  test("mergeExecutorTemplate should merge template with runtime configuration") {
+    val template = JobSubmitRequestItem(
+      priority = 0.5,
+      namespace = "template-namespace",
+      annotations = Map("template-annotation" -> "template-value"),
+      labels = Map("template-label" -> "template-value"),
+      podSpec = Some(PodSpec())
+    )
+
+    val cliConfig = armadaClientApp.CLIConfig(
+      queue = Some("test-queue"),
+      jobSetId = Some("test-job-set"),
+      namespace = Some("runtime-namespace"),
+      priority = Some(3.0),
+      containerImage = Some("spark:3.5.0"),
+      podLabels = Map.empty,
+      driverLabels = Map.empty,
+      executorLabels = Map.empty,
+      armadaClusterUrl = Some("armada://localhost:50051"),
+      nodeSelectors = Map("node-type" -> "compute"),
+      nodeUniformityLabel = Some("zone"),
+      executorConnectionTimeout = Some(300.seconds),
+      driverResources = armadaClientApp.ResourceConfig(
+        limitCores = Some("1"),
+        requestCores = Some("1"),
+        limitMemory = Some("1Gi"),
+        requestMemory = Some("1Gi")
+      ),
+      executorResources = armadaClientApp.ResourceConfig(
+        limitCores = Some("1"),
+        requestCores = Some("1"),
+        limitMemory = Some("1Gi"),
+        requestMemory = Some("1Gi")
+      )
+    )
+
+    val armadaJobConfig = armadaClientApp.ArmadaJobConfig(
+      queue = "test-queue",
+      jobSetId = "test-job-set",
+      jobTemplate = None,
+      driverJobItemTemplate = None,
+      executorJobItemTemplate = Some(template),
+      cliConfig = cliConfig
+    )
+
+    val runtimeAnnotations = Map("runtime-annotation" -> "runtime-value")
+    val runtimeLabels      = Map("runtime-label" -> "runtime-value")
+    val javaOptEnvVars     = Seq(EnvVar().withName("SPARK_JAVA_OPT_0").withValue("-Xmx1g"))
+
+    val result = armadaClientApp.mergeExecutorTemplate(
+      template,
+      0,
+      armadaJobConfig,
+      runtimeAnnotations,
+      runtimeLabels,
+      javaOptEnvVars,
+      "driver-service",
+      7078,
+      Seq.empty[Volume],
+      Map("node-type" -> "compute"),
+      sparkConf
+    )
+
+    result.priority shouldBe 3.0
```

Comment:
I really don't want to have to think about where this 3.0 came from.  If it was called "RUNTIME_PRIORITY" I would know.